### PR TITLE
feat(billing): implement stripe checkout

### PR DESCRIPTION
**Pull request by ygrishajev**
refs #247

### CODE REVIEW

**Review by baktun14 — APPROVED** [review]
Pretty cool! Can't wait to test it out

File: apps/api/drizzle/0005_parallel_outlaw_kid.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS "checkout_sessions" (
```
Comment (ygrishajev) [inline apps/api/drizzle/0005_parallel_outlaw_kid.sql]: need to think on better logic to ensure just one event is being processed

File: apps/api/drizzle/0006_sudden_namorita.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE "user_wallets" ALTER COLUMN "deployment_allowance" SET DATA TYPE integer;--> statement-breakpoint
```
Comment (ygrishajev) [inline apps/api/drizzle/0006_sudden_namorita.sql]: check if fractional values are set properly

Comment (ygrishajev) [reply apps/api/drizzle/0006_sudden_namorita.sql]: try updating numeric(10, 2) to higher digits

File: apps/api/src/billing/services/refill/refill.service.ts [hunk 3]
```diff
@@ -14,51 +15,54 @@ export class RefillService {
...
-      } catch (error) {
-        this.logger.error({ event: "REFILL_ERROR", error });
-      } finally {
-        await this.refillAll();
+      const { errors } = await PromisePool.withConcurrency(this.config.ALLOWANCE_REFILL_BATCH_SIZE)
+        .for(wallets)
+        .process(async wallet => this.refillWalletFees(wallet));
+
+      if (errors.length) {
+        this.logger.error({ event: "WALLETS_REFILL_ERROR", errors });
```
Comment (ygrishajev) [inline apps/api/src/billing/services/refill/refill.service.ts]: report failed wallets to sentry

File: apps/api/src/billing/services/stripe/stripe.service.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+      ],
+      mode: "payment",
+      customer: options.customerId,
+      success_url: `${options.redirectUrl}?session_id={CHECKOUT_SESSION_ID}&payment-success=true`,
+      cancel_url: `${options.redirectUrl}?session_id={CHECKOUT_SESSION_ID}&payment-canceled=true`
+    });
+  }
+
+  async charge(customerId: string, priceId: string) {
```
Comment (ygrishajev) [inline apps/api/src/billing/services/stripe/stripe.service.ts]: remove this one

File: apps/deploy-web/src/services/managed-wallet-http/managed-wallet-http.service.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    this.getWallet = debounce(this.getWallet.bind(this));
+  }
+
+  private extractSessionResults() {
+    const query = typeof window !== "undefined" && new URLSearchParams(window.location.search);
+
+    if (!query) {
+      return;
+    }
+    console.log('DEBUG query.get("payment-canceled")', query.get("payment-canceled"));
```
Comment (ygrishajev) [inline apps/deploy-web/src/services/managed-wallet-http/managed-wallet-http.service.ts]:
```suggestion
```

File: apps/deploy-web/src/services/managed-wallet-http/managed-wallet-http.service.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  private getWalletListParams(userId: string) {
+    const params: { userId: string; awaitSessionId?: string } = { userId };
+
+    if (this.checkoutSessionId) {
+      params.awaitSessionId = this.checkoutSessionId;
+    }
+    return params;
+  }
+
+  private clearSessionId() {
```
Comment (ygrishajev) [inline apps/deploy-web/src/services/managed-wallet-http/managed-wallet-http.service.ts]:
```suggestion
  private clearSessionResults() {
```

**Review by ygrishajev — COMMENTED** [review]
we need to disable start trial btn while it's in progress

**Review by baktun14 — APPROVED** [review]

**Pull request merged**